File: 1/2021.03.27/538.Convert BST to Greater Tree/python/solution.py
class Solution:
    def convertBST(self, root: TreeNode) -> TreeNode:
        def inorder(root):
            if root:
                inorder(root.right)
                root.val += self.sum
                self.sum = root.val
                inorder(root.left)
        
        self.sum = 0
        inorder(root)
        return root

File: 1/2021.03.27/538.Convert BST to Greater Tree/java/solution.java
class Solution {
    int sum = 0;
    public TreeNode convertBST(TreeNode root) {
        inorder(root);
        return root;
    }

    private void inorder(TreeNode root) {
        if (root != null) {
            inorder(root.right);
            sum += root.val;
            root.val = sum;
            inorder(root.left);
        }
    }
}
